File: Palestra di Dennis Ritchie (IMP)/Level Check/team11/LevelCheck.py
def is_even(level):
    return (level % 2 == 0)

def sort_pokemon_levels(pk_levels):
    odd_levels = []
    even_levels = []

    for level in pk_levels:
        if(is_even(level)):
            even_levels.append(level)
        else:
            odd_levels.append(level)
    return odd_levels, even_levels
    
def main():
    pk_levels = [1,4,7,9,15,30,56,47,100]
    odd_levels, even_levels = sort_pokemon_levels(pk_levels)
    print("Odd levels: ",odd_levels)
    print("Even levels: ",even_levels)

main()

File: Palestra di Dennis Ritchie (IMP)/Level Check/team3/LevelCheck.java
package BattleAndTrading;

public class LevelCheck {

    public static boolean isEven(int level) {return level %2 == 0;}

    public static int[][] sortLevels(int[] levels) {
        int evenCount = 0;
        int oddCount = 0;

        for (int level : levels) {
            if (isEven(level)) {evenCount++;}
            else
            {oddCount++;}
        }

        int[] evenLevels = new int[evenCount];
        int[] oddLevels = new int[oddCount];

        int evenIndex = 0;
        int oddIndex = 0;

        for (int level : levels) {
            if (isEven(level)) {evenLevels[evenIndex++] = level;}
            else
            {oddLevels[oddIndex++] = level;}
        }

        return new int[][]{evenLevels, oddLevels};
    }

    public static void main(String[] args) {
        int[] levels = {5, 8, 13, 24, 7, 18, 9, 14};

        int[][] sortedLevels = sortLevels(levels);

        System.out.print("Even levels: [");
        for (int level : sortedLevels[0]) {System.out.print(level + " ");}
        System.out.print("]\n");

        System.out.print("Odd levels: [");
        for (int level : sortedLevels[1]) {System.out.print(level + " ");}
        System.out.print("]");
    }
}